experiments/ClimaEarth/user_io/arg_parsing.jl

```diff
@@ -0,0 +1,248 @@
+"""
+    get_coupler_config()
+
+Reads in the configuration file and job ID from the command line.
+A dictionary is constructed from the input configuration file, and this
+is returned along with the job ID.
+
+# Returns
+- `config_dict`: A dictionary mapping configuration keys to the specified settings
+"""
+function get_coupler_config()
+    # Read in command line arguments
+    parsed_args = parse_commandline(argparse_settings())
+
+    # Extract the configuration file and job ID
+    config_file = parsed_args["config_file"]
+    job_id = parsed_args["job_id"]
+    # Get the job ID from the config file string if not provided
+    job_id = isnothing(job_id) ? string(split(split(config_file, '/')[end], '.')[1]) : job_id
+
+    # Read in config dictionary from file, overriding the defaults in `parsed_args`
+    config_dict = merge(parsed_args, YAML.load_file(parsed_args["config_file"]))
+    config_dict["job_id"] = job_id
+    return config_dict
+end
+
+"""
+    get_coupler_args!(config_dict)
+
+Extract the necessary arguments from the coupled configuration dictionary.
+This function may modify the input dictionary to remove unnecessary keys.
```

Comment:
Let's split this into two: a function that removes the unnecessary keys, and one that returns the the coupler configs. 

Having mutating functions that return stuff adds complexity that can be easily avoided.